pkg/notifier/beans/beans.go

```diff
@@ -0,0 +1,394 @@
+	Providers    []*Provider       `json:"providers"`
+}
+
+func (notificationSettingsRequest *NotificationConfigRequest) GenerateSettingCombinationsV1() []*LocalRequest {
+
+	var tempRequest []*LocalRequest
+	if len(notificationSettingsRequest.TeamId) == 0 && len(notificationSettingsRequest.EnvId) == 0 && len(notificationSettingsRequest.AppId) > 0 {
+		for _, item := range notificationSettingsRequest.AppId {
+			tempRequest = append(tempRequest, &LocalRequest{AppId: item})
+		}
+	} else if len(notificationSettingsRequest.TeamId) == 0 && len(notificationSettingsRequest.EnvId) > 0 && len(notificationSettingsRequest.AppId) == 0 {
+		for _, item := range notificationSettingsRequest.EnvId {
+			tempRequest = append(tempRequest, &LocalRequest{EnvId: item})
+		}
+	} else if len(notificationSettingsRequest.TeamId) > 0 && len(notificationSettingsRequest.EnvId) == 0 && len(notificationSettingsRequest.AppId) == 0 {
+		for _, item := range notificationSettingsRequest.TeamId {
+			tempRequest = append(tempRequest, &LocalRequest{TeamId: item})
+		}
+	} else if len(notificationSettingsRequest.TeamId) == 0 && len(notificationSettingsRequest.EnvId) > 0 && len(notificationSettingsRequest.AppId) > 0 {
+		for _, itemE := range notificationSettingsRequest.EnvId {
+			for _, itemA := range notificationSettingsRequest.AppId {
+				tempRequest = append(tempRequest, &LocalRequest{EnvId: itemE, AppId: itemA})
+			}
+		}
+	} else if len(notificationSettingsRequest.TeamId) > 0 && len(notificationSettingsRequest.EnvId) > 0 && len(notificationSettingsRequest.AppId) == 0 {
+		for _, itemT := range notificationSettingsRequest.TeamId {
+			for _, itemE := range notificationSettingsRequest.EnvId {
+				tempRequest = append(tempRequest, &LocalRequest{TeamId: itemT, EnvId: itemE})
+			}
+		}
+	} else if len(notificationSettingsRequest.TeamId) > 0 && len(notificationSettingsRequest.EnvId) == 0 && len(notificationSettingsRequest.AppId) > 0 {
+		for _, itemT := range notificationSettingsRequest.TeamId {
+			for _, itemA := range notificationSettingsRequest.AppId {
+				tempRequest = append(tempRequest, &LocalRequest{TeamId: itemT, AppId: itemA})
+			}
+		}
+	} else if len(notificationSettingsRequest.TeamId) > 0 && len(notificationSettingsRequest.EnvId) > 0 && len(notificationSettingsRequest.AppId) > 0 {
+		for _, itemT := range notificationSettingsRequest.TeamId {
+			for _, itemE := range notificationSettingsRequest.EnvId {
+				for _, itemA := range notificationSettingsRequest.AppId {
+					tempRequest = append(tempRequest, &LocalRequest{TeamId: itemT, EnvId: itemE, AppId: itemA})
+				}
+			}
+		}
+	} else {
+		tempRequest = append(tempRequest, &LocalRequest{PipelineId: notificationSettingsRequest.PipelineId})
+	}
+
+	return tempRequest
+}
+
+// GetIdsByTypeIndex
+// if new criteria fields are added, add a case to this function
+func (notificationConfigRequest *NotificationConfigRequest) GetIdsByTypeIndex(index selectorIndex) []*int {
```

Comment:
this func seems like part of a util file

Reply:
same here